src/mavlink/mavlink_camera.rs

```diff
@@ -848,17 +828,20 @@ fn camera_capture_status() -> mavlink::common::MavMessage {
     )
 }
 
-fn video_stream_information(
-    video_name: &str,
-    information: &MavlinkCameraInformation,
-) -> mavlink::common::MavMessage {
-    let name_str = String::from(video_name);
+fn video_stream_information(information: &MavlinkCameraInformation) -> mavlink::common::MavMessage {
+    let source_string: Vec<char> = information
+        .video_source_type
+        .inner()
+        .source_string()
+        .chars()
+        .collect();
     let mut name: [char; 32] = ['\0'; 32];
-    for index in 0..name_str.len() as u32 {
-        name[index as usize] = name_str.as_bytes()[index as usize] as char;
-    }
+    let len = std::cmp::min(source_string.len(), name.len());
+    name[..len].copy_from_slice(&source_string[..len]);
 
-    let uri: Vec<char> = format!("{}\0", information.video_stream_uri).chars().collect();
+    let uri: Vec<char> = format!("{}\0", information.video_stream_uri)
```

Comment:
I think this is one of a few cases where it seems a bit inconsistent. So far I found many similar occasions where I initially could freely choose to break or not into several lines and it will not complain, and suddenly by changing things around it starts to be less flexible... Anyway, I feel that I need to just accept its temperamental ruling :laughing: